npm/metrics/ai-utils.go

```diff
@@ -0,0 +1,91 @@
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+var (
+	th aitelemetry.TelemetryHandle
+)
+
+// CreateTelemetryHandle creates
+func CreateTelemetryHandle(version, aiMetadata string) error {
+
+	aiConfig := aitelemetry.AIConfig{
+		AppName:                   util.AzureNpmFlag,
+		AppVersion:                version,
+		BatchSize:                 util.BatchSize,
+		BatchInterval:             util.BatchInterval,
+		RefreshTimeout:            util.RefreshTimeout,
+		DebugMode:                 util.DebugMode,
+		GetEnvRetryCount:          util.GetEnvRetryCount,
+		GetEnvRetryWaitTimeInSecs: util.GetEnvRetryWaitTimeInSecs,
+	}
+
+	var err error
+	th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
+
+	for i := 0; err != nil && i < util.AiInitializeRetryCount; i++ {
+		log.Logf("Failed to init AppInsights with err: %+v for %d time", err, i + 1)
+		time.Sleep(time.Minute * 5)
```

Comment:
This sleeps for 5 minutes on every iteration, blocking the main function if I read correctly. That may be too long to sleep.

Reply:
I moved this initialization logic from npm.go from line 117 to 121. It is in a separate go routine previously. How about I keeep this retry logic but move it to a separate go routine?